mautic/mautic
# Include companies in contact related data sources

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR adds an ability to include companies in contact related data sources:

- Asset downloads
- Campaign events
- Message queue
- Emails sent
- Form submissions
- All contacts
- Mobile notifications
- Page hits

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
--

#### Steps to test this PR:
0. *Do not checkout this branch unless you have completed first 2 steps :)*
1. Create reports for all data sources mentioned in description of this task. Make sure you have data in all types of report. I would recommend to include contact ID in columns.
2. Make sure that there is at least 1 contact which more than 1 company in each type of report (better are more contacts with several companies).
3. Checkout this branch. Clear cache.
4. Check all reports that all data are same.
5. Go through each report and edit it - add new company related columns. Check the result of report - contact with 2 companies should be twice there.
6. Add a condition for primary company for each report. Contact with 2 companies shoul be presented only once in the result.

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2.
